api/app/services/llm_extraction/pre_filters.py

```
@@ -0,0 +1,309 @@
+
+    Args:
+        timeout_seconds: Maximum time allowed for regex execution
+
+    Returns:
+        Decorated function with timeout protection
+    """
+
+    def decorator(func: Callable) -> Callable:
+        @wraps(func)
+        def wrapper(*args, **kwargs):
+            def handler(signum, frame):
```

Ruff (0.14.11)

Unused function argument: `signum`

(ARG001)